09_class_bank_card.py

```diff
@@ -0,0 +1,261 @@
+        self.__card_num = int('4441' + str(random.randint(100000000000,
+                                                          999999999999)))
+        self.__balance = 0
+
+    def __str__(self):
+        """Human-readable string representation of the Credit card."""
+        return (f'This credit card is in the name of '
+                f'{self.name} {self.surname}\n'
+                f'Credit card number {self.__card_num}\n'
+                f'Card exp date {self.__exp_date}')
+
+    @classmethod
+    def validate_date(cls, end_date):
+        """Check the date is greater than or equal to current date."""
+        return end_date >= cls.CURRENT_DATE
+
+    @classmethod
+    def validate_length_pin(cls, pin):
+        """Check the pin code have a length of 4 characters long."""
+        return len(str(pin)) == cls.PIN_LENGTH
+
+    @staticmethod
+    def _uah_to_usd(uah_value):
+        """Convert USD to UAH."""
+        dollar_exchange_rate = 40.1
+        return uah_value * dollar_exchange_rate
+
+    @property
+    def pin_code(self):
+        """Getter for Pin code."""
+        return self.__pin
+
+    @pin_code.setter
+    def pin_code(self, pin):
+        """Setter for Pin code."""
+        self.__pin = pin
+
+    @property
+    def balance(self):
+        """Getter for credit card balance."""
+        return format(self.__balance, '.2f')
+
+    @property
+    def cvv_code(self):
+        """Getter for CVV code."""
+        return self.__cvv
+
+    @property
+    def card_number(self):
+        """Getter for Card number."""
+        return self.__card_num
+
+    @property
+    def exp_date(self):
+        """Getter for expiration card date."""
+        return self.__exp_date
+
+    def check_pin_code(self, current_pin):
+        """Check if pin code is correct."""
+        return current_pin == self.__pin
+
+    def check_cvv_code(self, cvv):
+        """Check if cvv code is correct."""
+        return cvv == self.__cvv
+
+    def check_card_number(self, current_card_num):
+        """Check if card number is correct."""
+        return current_card_num == self.__card_num
+
+    def validate_expired_card(self, end_date):
+        """Check the card expiration date."""
+        return self.validate_date(end_date)
+
+    def __income(self, new_sum, usd=False):
+        """Balance increase.
+
+        params new_sum: An integer representing the amount of money.
+        params usd: A bool representing if sum in usd.
+        """
+        operation = self._uah_to_usd(new_sum) if usd else new_sum
+        self.__balance += operation
+        return self.__balance
+
+    def __expenses(self, new_sum, usd=False):
+        """Decrease in balance. With balance check for negative value.
+
+        params new_sum: An integer representing the amount of money.
+        params usd: A bool representing if sum in usd.
+        """
+        operation = self._uah_to_usd(new_sum) if usd else new_sum
+        if self.__balance - operation >= 0:
```

Comment:
What if this condition fails? We should probably raise some error  or return False